ReferenceAppKotlin/app/src/main/java/com/android/tv/reference/playback/PlaybackFragment.kt

```diff
@@ -87,77 +86,96 @@ class PlaybackFragment : VideoSupportFragment() {
+            val forwardingPlayer = object : ForwardingPlayer(exoplayer) {
+                override fun stop() {
+                    // Treat stop commands as pause, this keeps ExoPlayer, MediaSession, etc.
+                    // in memory to allow for quickly resuming. This also maintains the playback
+                    // position so that the user will resume from the current position when backing
+                    // out and returning to this video
+                    Timber.v("Playback stopped at $currentPosition")
+                    // This both prevents playback from starting automatically and pauses it if
+                    // it's already playing
+                    playWhenReady = false
+                }
+            }
+            val activityPendingIntent = PendingIntent.getActivity(
+                requireContext(),
+                0,
+                Intent(requireContext(), MainActivity::class.java).apply {
+                    flags = Intent.FLAG_ACTIVITY_CLEAR_TASK
+                },
+                PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+            )
+            mediaSession =
+                MediaSession.Builder(/* context= */ requireContext(), forwardingPlayer)
+                    .setId(MEDIA_SESSION_TAG)
+                    .setSessionActivity(activityPendingIntent)
+                    .build()
+            CastReceiverContext.getInstance().mediaManager.setSessionCompatToken(
+                mediaSession!!.sessionCompatToken as MediaSessionCompat.Token)
+
+            val mediaMetadata =
+                MediaMetadata.Builder()
+                    .setArtworkUri(Uri.parse(video.thumbnailUri))
+                    .setTitle(video.name)
+                    .setDescription(video.description)
+                    .build()
+            val mediaItem =
+                MediaItem.Builder()
+                    .setUri(video.videoUri)
+                    .setMediaId(video.id)
+                    .setMediaMetadata(mediaMetadata)
+                    .build()
+            exoplayer.setMediaItem(mediaItem)
+            exoplayer.prepare()
+            prepareGlue(exoplayer)
+            viewModel.onStateChange(VideoPlaybackState.Load(video))
         }
-
-        viewModel.onStateChange(VideoPlaybackState.Load(video))
     }
 
-    private fun destroyPlayer() {
-        mediaSession.isActive = false
-        mediaSessionConnector.setPlayer(null)
-        exoplayer?.let {
-            // Pause the player to notify listeners before it is released.
-            it.pause()
-            it.release()
-            exoplayer = null
+    private fun releaseMediaSession() {
+        mediaSession?.run {
+            player.release()
+            release()
+            mediaSession = null
+            // TODO Remove comment after b/218604167 has been release in the Cast TV library.
+            // CastReceiverContext.getInstance().mediaManager.setSessionCompatToken(null)
```

Comment:
This call replaces the method I have removed from `CastHelper`/`CastHelperTest`. We just call it directly here without the indirection through the `CastHelper`. So we don't need to deal with the bug in the test case. The indirection in `CastHelper` was super trivial and a bit meaningless as it just calls through the cast method. We can put that back when the bug is fixed and we can uncomment the call here.